### PR TITLE
Feature/ci build netgear rax40

### PR DESCRIPTION
**Pull request by rmelotte**
Keeping it as a draft until we can confirm all tests from CI pass. 
Will also need to integrate changes from #616 .

### CODE REVIEW

File: tools/docker/builder/openwrt/build.sh [hunk 1]
```diff
@@ -26,6 +26,7 @@ build_image() {
            --build-arg OPENWRT_VERSION \
            --build-arg TARGET \
            --build-arg PRPL_FEED \
+	   --build-arg PRPLMESH_VARIANT \
```
Comment (arnout) [inline tools/docker/builder/openwrt/build.sh]: Yet another tab problem...

File: tools/docker/builder/openwrt/build.sh [hunk 2]
```diff
@@ -86,6 +87,8 @@ main() {
     dbg "IMAGE_ONlY=IMAGE_ONLY"
     dbg "TARGET_DEVICE=$TARGET_DEVICE"
     dbg "TAG=$TAG"
+    dbg "TARGET=$TARGET"
```
Comment (arnout) [inline tools/docker/builder/openwrt/build.sh]: This is an unrelated change and I'm not sure if it's very relevant.

File: tools/docker/builder/openwrt/scripts/build.sh [hunk 3]
```diff
@@ -7,5 +7,5 @@ cp -r /home/openwrt/prplMesh_source /home/openwrt/prplMesh
 rm -rf /home/openwrt/prplMesh/build
 
 make package/feeds/prpl/prplmesh/prepare USE_SOURCE_DIR="/home/openwrt/prplMesh" V=s
-make package/feeds/prpl/prplmesh/compile V=sc -j"$(nproc)"
-find bin -name 'prplmesh_*.ipk' -exec cp -v {} "prplmesh-${TARGET}-${OPENWRT_VERSION}-${PRPLMESH_VERSION}.ipk" \;
+make package/feeds/prpl/prplmesh/compile VARIANT="$PRPLMESH_VARIANT" V=sc -j"$(nproc)"
```
Comment (arnout) [inline tools/docker/builder/openwrt/scripts/build.sh]: This shouldn't be needed since the variant is specified in the config, right? 

Also, since it's an ARG and not an ENV, is it even saved in the environment when running the container?

File: tools/docker/builder/openwrt/Dockerfile [hunk 4]
```diff
@@ -41,7 +44,12 @@ RUN git clone "$OPENWRT_REPOSITORY" openwrt \
 
 WORKDIR /home/openwrt/openwrt
 
-RUN echo "CONFIG_SDK=y" > .config \
+RUN if [ -n "$INTEL_FEED" ] ; then \
+	echo "src-git intel $INTEL_FEED" >> feeds.conf.default \
+	&& scripts/feeds update intel \
+	&& scripts/feeds install intel_mips ; \
```
Comment (arnout) [inline tools/docker/builder/openwrt/Dockerfile:50]: Can you explain better in the commit message why it is necessary to explicitly install this one?

File: tools/docker/builder/openwrt/build.sh [hunk 5]
```diff
@@ -10,7 +10,6 @@ usage() {
     echo "  options:"
     echo "      -h|--help - show this help menu"
     echo "      -d|--target-device the device to build for"
-    echo "      -f|--prpl-feed the prpl feed to use"
```
Comment (arnout) [inline tools/docker/builder/openwrt/build.sh:14]: Can you extend the help text with the environment variables that affect the build? I.e., PRPL_FEED and INTEL_FEED. And include their defaults as well.

File: tools/docker/builder/openwrt/build.sh [hunk 6]
```diff
@@ -64,6 +64,12 @@ main() {
                     turris-omnia)
                         TARGET=mvebu
                         ;;
+                    netgear-rax40)
+                        TARGET=intel_mips
+                        PRPLMESH_VARIANT="-dwpal"
+                        PRPL_FEED="https://git.prpl.dev/prplmesh/iwlwav.git^e9ee6b109764dbb91b6cb399ccc0f78e5b04e260"
```
Comment (arnout) [inline tools/docker/builder/openwrt/build.sh]: Is this really still needed? Can't we still use this feed by specifying the variant?

I'm running a test build, if successful I'll push that with fixup commits.

**Review by arnout — APPROVED** [review]
Easily fixed comments only.

I did a rebase on master to be able to test some things, so I'll force-push that after running a test build.

**Comment by arnout** [comment]
Build didn't complete yet but I had to go home, so let's wait for CI to report if it works or not...

File: tools/docker/builder/openwrt/scripts/build.sh (hunk 3 above)
Comment (rmelotte) [reply tools/docker/builder/openwrt/scripts/build.sh]: Last time I tried it didn't use the correct variant even when it was selected in the config file.

This was an attempt to force it to use the correct variant, but it was a bad idea: `VARIANT` will be the same for all the packages, which means that the hostapd variant for example, will be `-dwpal` or `-nl80211`...

Removing.

File: tools/docker/builder/openwrt/build.sh (hunk 6 above)
Comment (rmelotte) [reply tools/docker/builder/openwrt/build.sh]: When using the latest feed for the nl80211 targets, dwpal gets compiled although it should not (the intel feed is missing so it fails with safeclibs missing).
I suggest we first merge this PR using the 2 separate feed versions and open an issue about it.
That way we can finally build the dwpal variant in CI and at least spot potential compilation issues introduced by other changes early.

File: tools/docker/builder/openwrt/Dockerfile (hunk 4 above)
Comment (rmelotte) [reply tools/docker/builder/openwrt/Dockerfile:50]: Done.

File: tools/docker/builder/openwrt/build.sh (hunk 5 above)
Comment (rmelotte) [reply tools/docker/builder/openwrt/build.sh:14]: Done.

**Comment by rmelotte** [comment]
For reference, latest pipeline was here: https://gitlab.com/prpl-foundation/prplMesh/pipelines/106383852
It still fails because it compiles dwpal even for the nl80211 targets.
I'll force push again with the 2 different feeds version, and the changes that were requested. 

I kept a copy of the 2 additional fixup commits in case we still want them.

**Comment by rmelotte** [comment]
I changed the value of `INTEL_FEED` to be an empty string (to make it obvious that it's being used in the script, like we do for the other variables). Because of this, the CI needs to rebuild the whole image.

I had to cancel the latest one to avoid compiling the same image twice at the same time (which is why github shows the pipeline as canceled).

Marking it as ready to review already, but we will need to re-run the pipeline once the current build is done.

**Review by arnout — APPROVED** [review]

**Comment by rmelotte** [comment]
The build failed because of a container name clash, so I added an additional commit to use unique container names (image names were already unique, containers names were only timestamped).

I also noticed the containers weren't always removed, so I added another commit to make sure containers are removed even when the build fails.

Finally, I also fixed some debug messages which printed wrong values.

**Comment by rmelotte** [comment]
Added quotes to the target in gitlab-ci.yml, since every other target are quoted.

**Comment by rmelotte** [comment]
Rebased on master.
